Docs for scaling down and creating silences

T-Kukawka opened the pull request:
towards:
https://github.com/giantswarm/giantswarm/issues/19503
https://github.com/giantswarm/roadmap/issues/546

whites11 commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/cronjob-scale-down-silence.yaml:
```diff
@@ -0,0 +1,48 @@
+  labels:
+    app: silence-customer
+spec:
+  schedule: "47 16 * * 5" #Timezone - UTC by default - Fridays, 16;47 UTC
```
```suggestion
  schedule: "47 16 * * 5" #Timezone - UTC by default - Fridays, 16:47 UTC
```

whites11 commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+    parent: cost-optimization
+last_review_date: 2021-11-15
+user_questions:
+- How can i scale down cluster nodes to 0?
```
```suggestion
- How can I scale down cluster nodes to 0?
```

whites11 commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+            - -c
+            - |
+
+              kubectl patch MachinePool -n {YOUR_ORGANIZATION} {CLUSTER-ID} --type merge -p '{"metadata" : {"annotations": {"cluster.k8s.io/cluster-api-autoscaler-node-group-min-size": "1"}}}'
```
Maybe I would use a more reasonable min value such as 3

T-Kukawka replied:
this is just as example - can be set by customer to whatever they have currently

marians commented:
Here is a preview:

![image](https://user-images.githubusercontent.com/273727/142006772-8d44cbc0-67dc-4601-a6c0-f9fa71fca247.png)

The `linkTitle` attribute is shown in the menu and as a title in the list of sub-pages. Would be cool to have those blanks back :)

marians commented on src/content/advanced/cost-optimization/cost-optimization-tools/index.md:
```diff
@@ -1,11 +1,11 @@
 ---
-linkTitle: Cost-optimization
+linkTitle: Cost-optimization-tools
```
```suggestion
linkTitle: Tools
```

The short title (linkTitle) can omit any repitition from parent titles. It only has to work in this context.

marians commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+---
+linkTitle: Workload-clusters-scale-down
```
```suggestion
linkTitle: Scaling down
```

marians commented:
The idea of offering YAML files for download is neat, but it's something we don't do elsewhere. And I'm thinking there are some downsides:

1. Its very easy to forget those YAML files when it comes to keeping the content up-to-date.
2. As a user, it would be easy to miss the download, but it would be relatively simple to copy the displayed example instead.

(1) alone is reason for me to advice against the additional download file. Because of (2) the benefit would not be huge anyway.

marians commented:
Can we please change the folder name from `cost-optimization-tools` to `tools`?

marians commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+kind: CronJob
+metadata:
+  name: silences-customer-delete-{CLUSTER-ID}
+  namespace: {YOUR_ORGANIZATION}
```
```suggestion
  namespace: org-{YOUR_ORGANIZATION}
```

marians replied:
Same in other CRs

marians commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
````diff
@@ -0,0 +1,199 @@
+
+You can see the example of such Custom Resource below:
+
+```yaml
+apiVersion: monitoring.giantswarm.io/v1alpha1
+kind: Silence
+metadata:
+  name: INSTALLATION-CLUSTER-ID-silence-cluster
+spec:
+  targetTags:
+  - name: installation
+    value: YOUR_INSTALLATION_NAME
+  matchers:
+  - name: cluster_id
+    value: CLUSTER_ID
+    isRegex: false
+```
````
In the Management API, there is no such thing as a cluster ID. And in the new incarnation of the UIs there isn't either. What used to be called the cluster ID is now the cluster name.

When in doubt, explain that the cluster name to use here is the name of the `Cluster` resource and the unique identifier of the cluster.

T-Kukawka commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/cronjob-scale-down-silence.yaml:
```diff
@@ -0,0 +1,48 @@
+              EOF
+
+              # patch annotations of min and max values for autoscaling to 0
+              kubectl patch MachinePool -n {YOUR_ORGANIZATION} {CLUSTER_ID} --type merge -p '{"metadata" : {"annotations": {"cluster.k8s.io/cluster-api-autoscaler-node-group-min-size": "0"}}}'
```
setting min and max to 0 will result in deleting the machines - yes

marians commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+---
+linkTitle: Scaling down
+title: Workload Clusters Scale Down
```
```suggestion
title: Scaling down workload clusters
```

Appears more natural to me. Also we use sentence case for headlines.

marians commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+---
+linkTitle: Scaling down
+title: Workload Clusters Scale Down
+description: In order to save costs over the weekends or stale periods some entities would like to scale down the clusters to 0 nodes, this guide describes such process for Giant Swarm clusters.
```
```suggestion
description: In order to save costs over the weekends or stale periods some entities would like to scale down the clusters to zero nodes, this guide describes such process for Giant Swarm clusters.
```

Numbers one to ten should be spelled out in text

marians commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/index.md:
```diff
@@ -0,0 +1,199 @@
+  - https://github.com/orgs/giantswarm/teams/team-phoenix
+---
+
+# Periodically scaling down workload clusters to 0 worker nodes
```
```suggestion
# Periodically scaling down workload clusters to zero worker nodes
```

fiunchinho commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/cronjob-scale-down-silence.yaml:
```diff
@@ -0,0 +1,48 @@
+                  isRegex: false
+              EOF
+
+              # patch annotations of min and max values for autoscaling to 0
```
Maybe we should say `YOUR_ORGANIZATION_NAMESPACE` rather than `YOUR_ORGANIZATION`? Also, we need the nodepool ID rather than the CLUSTER_ID

fiunchinho commented on src/content/advanced/cost-optimization/workload-clusters-scale-down/cronjob-silence-rbac.yaml:
```diff
@@ -0,0 +1,68 @@
+    - 'configMap'
+    - 'secret'
+    - 'hostPath'
+  allowPrivilegeEscalation: false
```
```suggestion
  allowPrivilegeEscalation: false

```

T-Kukawka commented:
I have addressed all comments - i went with Jose suggestion of {YOUR_ORGANIZATION_NAMESPACE} clarification that seems more 'clean'

I have removed the files and their references, renamed the folder - i hope i did not miss anything

T-Kukawka commented:
👋 @marians 🙏
